libs/designsystem/src/lib/components/popover/popover.component.ts

```diff
@@ -0,0 +1,210 @@
+import { DOCUMENT } from '@angular/common';
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output,
+  Renderer2,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+
+import { DesignTokenHelper } from '../../helpers';
+
+export enum HorizontalDirection {
+  right = 'right',
+  left = 'left',
+}
+@Component({
+  selector: 'kirby-popover',
+  template: `
+    <div #wrapper class="wrapper"><ng-content></ng-content></div>
+  `,
+  styleUrls: ['./popover.component.scss'],
+})
+export class PopoverComponent implements AfterViewInit, OnChanges, OnDestroy {
+  @ViewChild('wrapper', { static: true, read: ElementRef })
+  wrapperElement: ElementRef<HTMLDivElement>;
+
+  @Input()
+  target: HTMLElement | ElementRef<HTMLElement>;
+
+  @Output()
+  willHide = new EventEmitter<void>();
+
+  private targetElement: HTMLElement;
+
+  private isShowing = false;
+
+  @Input()
+  popout: HorizontalDirection = HorizontalDirection.right;
+
+  @HostListener('click')
+  _backdropClick() {
+    this.willHide.emit();
+    this.hide();
+  }
+
+  @HostListener('window:resize')
+  _onWindowResize() {
+    if (this.isShowing) {
+      this.willHide.emit();
+      this.hide();
+    }
+  }
+
+  constructor(
+    private elementRef: ElementRef<HTMLElement>,
+    @Inject(DOCUMENT) private document: Document,
+    private renderer: Renderer2
+  ) {}
+
+  private zIndex = parseInt(DesignTokenHelper.zLayer('popover'));
+
+  ngAfterViewInit(): void {
+    const ionModal = this.elementRef.nativeElement.closest('ion-modal');
+    if (ionModal) {
+      this.zIndex = parseInt(ionModal.style.zIndex) + 1;
+      this.elementRef.nativeElement.style.zIndex = `${this.zIndex}`;
+    }
```

Comment:
Consider throwing this into a function explaining what is going on such as `placeAboveNearestModal`